### DISCUSSION TITLE
More configurable options for ECS agent

### DISCUSSION THREAD

**Discussion by mbaciu-gpsw** [bottlerocket-os/bottlerocket#2480] [Q&A]
Hi all,

searching through docs, discussions, PRs and code, looks like bottlerocket exposes only a subset of the available configuration options for the **ecs agent** (eg `logging-drivers`, `allow-privileged-containers`, `loglevel`, `enable-spot-instance-draining` and `image-pull-behavior`). While all these are quite useful for us , there are still some configs for which we have a use case, and not all of them have an alternative (like eg addresing them at the task definition level).

Here are some of these ecs agent configs that we're missing right now in bottlerocket:

- `ECS_TASK_METADATA_RPS_LIMIT`:  we've been bumping this (as advised by AWS support), after noticing 429s returned by the ecs agent when hitting the /v2/credentials endpoint
- `ECS_CONTAINER_STOP_TIMEOUT`: we've used this with some containers that took a little more time to terminate that the default 30 seconds. Helped with services ending up unnecessarily redoing jobs that were probably very close to finish.
- `ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION`: helped us with the host volume being filled with data used by stopped tasks. 
- `ECS_RESERVED_MEMORY`:  unless I'm wrong, this is set to the default 0 in bottlerocket. AWS documentation suggest to _[also reserve some memory for the Amazon ECS container agent](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/memory-management.html)_. We've also been using it.

My questions: is configuring above supported with the ecs variant? And if not, is this something that could be considered for a new feature?

Thanks!

**Comment by arnaldo2792** [comment]
Hi, currently there isn't a way to configure those features in Bottlerocket's API system. I created an issue to track the implementation of them:

https://github.com/bottlerocket-os/bottlerocket/issues/2483
